### PR TITLE
required field error fixed

### PR DESCRIPTION
**Pull request by adnan1naeem**
Description
 changes validation color to red which was grey before
Related Issue
https://github.com/BlueEastCode/mevris-client-plugin-user-management/issues/46
Screenshot 2019-09-20 at 3 56 02 PM

### CODE REVIEW

**Comment by adnan1naeem** [comment]
@artalat  please merge

**Comment by artalat** [comment]
This is not the correct fix. Please discuss with me today.

File: src/JsonGraphqlForm/__tests__/JsonGraphqlForm.graphql.test.tsx [hunk 1]
```diff
@@ -131,11 +131,11 @@ describe('JsonGraphqlForm', () => {
 			// Post Validation
 			firstName = wrapper.find('TextInput[name="firstName"]').first();
 			expect(firstName.prop('error')).toBe(true);
-			expect(firstName.prop('helperText')).toBe('I m sorry, but we dont like your name.');
```
Comment (artalat) [inline src/JsonGraphqlForm/__tests__/JsonGraphqlForm.graphql.test.tsx]: why was this test removed?

File: src/JsonGraphqlForm/__tests__/JsonGraphqlForm.graphql.test.tsx [hunk 2]
```diff
@@ -117,7 +117,7 @@ describe('JsonGraphqlForm', () => {
 
 			// Pre Validation
 			let firstName: any = wrapper.find('TextInput[name="firstName"]').first();
-			expect(firstName.prop('error')).toBe(undefined);
+			expect(firstName.prop('error')).toBe(undefined);;
```
Comment (artalat) [inline src/JsonGraphqlForm/__tests__/JsonGraphqlForm.graphql.test.tsx]: why was this change made?

File: src/JsonGraphqlForm/__tests__/JsonGraphqlForm.graphql.test.tsx [hunk 3]
```diff
@@ -131,11 +131,11 @@ describe('JsonGraphqlForm', () => {
 			// Post Validation
 			firstName = wrapper.find('TextInput[name="firstName"]').first();
 			expect(firstName.prop('error')).toBe(true);
-			expect(firstName.prop('helperText')).toBe('I m sorry, but we dont like your name.');
+
 
 			password = wrapper.find('TextInput[name="password"]').first();
 			expect(password.prop('error')).toBe(true);
-			expect(password.prop('helperText')).toBe('The password cannot be less than 8 characters');
```
Comment (artalat) [inline src/JsonGraphqlForm/__tests__/JsonGraphqlForm.graphql.test.tsx]: why was this test removed?

File: src/JsonForm/Fields/FormTextInput.tsx [hunk 4]
```diff
@@ -34,29 +41,39 @@ const validate = (props: FormTextInputProps) => (value: string) => {
...
+	);
+};
 
 FormTextInput.defaultProps = {
 	MainComponent: TextInput
-};
\ No newline at end of file
+};
+
+FormTextInput.defaultStyles = (_theme: Theme) => ({
```
Comment (artalat) [inline src/JsonForm/Fields/FormTextInput.tsx]: what is the purpose of this?

File: src/JsonForm/Fields/FormTextInput.tsx [hunk 5]
```diff
@@ -34,29 +41,39 @@ const validate = (props: FormTextInputProps) => (value: string) => {
...
-				},
-			};
-
-			return (<BaseFormField {...inputProps} />);
-		}}
-	</Field>
-);
+export const FormTextInput = (props: FormTextInputProps) => {
+
+	return (
```
Comment (artalat) [inline src/JsonForm/Fields/FormTextInput.tsx]: revert to previous code style that doesnt have a return statement

File: src/JsonForm/Fields/FormTextInput.tsx [hunk 6]
```diff
@@ -1,18 +1,25 @@
...
 import { Field } from 'formik';
 import React from 'react';
 import { TextInputProps } from '@bluebase/components';
-import { getComponent } from '@bluebase/core';
+import { ViewStyle } from 'react-native';
 
 const BaseFormField = getComponent<BaseFormFieldProps>('BaseFormField');
 const TextInput = getComponent<TextInputProps>('TextInput');
 
+export interface FormTextInputStyles {
```
Comment (artalat) [inline src/JsonForm/Fields/FormTextInput.tsx]: why do we need this?

File: src/JsonForm/Fields/FormTextInput.tsx [hunk 7]
```diff
@@ -1,18 +1,25 @@
...
+
+	root?: ViewStyle
+}
 export type FormTextInputProps<T = {}> = TextInputProps & BaseFormFieldProps & T & {
 	children?: React.ReactNode;
 	validate?: ((value: any) => string | Promise<void> | undefined);
 	name: string;
 	type?: string;
 	value?: any;
+	styles?: FormTextInputStyles;
```
Comment (artalat) [inline src/JsonForm/Fields/FormTextInput.tsx]: why do we need this?

File: src/JsonForm/BaseFormField.tsx [hunk 8]
```diff
@@ -52,7 +53,7 @@ export class BaseFormField extends React.PureComponent<BaseFormFieldProps> {
 				{left && <View style={styles.leftContainer} testID="base-form-field-left">{left}</View>}
 				{MainComponent &&
 					<View style={styles.mainContainer} testID="base-form-field-main">
-						<MainComponent {...props}>
+						<MainComponent   {...props}>
```
Comment (artalat) [inline src/JsonForm/BaseFormField.tsx]: why was this change made?

**Review by artalat — CHANGES_REQUESTED** [review]

File: src/JsonGraphqlForm/__tests__/JsonGraphqlForm.graphql.test.tsx (hunk 3 above)
Comment (adnan1naeem) [reply src/JsonGraphqlForm/__tests__/JsonGraphqlForm.graphql.test.tsx]: i have reverted the code

File: src/JsonGraphqlForm/__tests__/JsonGraphqlForm.graphql.test.tsx (hunk 2 above)
Comment (adnan1naeem) [reply src/JsonGraphqlForm/__tests__/JsonGraphqlForm.graphql.test.tsx]: I have fixed it

File: src/JsonGraphqlForm/__tests__/JsonGraphqlForm.graphql.test.tsx (hunk 1 above)
Comment (adnan1naeem) [reply src/JsonGraphqlForm/__tests__/JsonGraphqlForm.graphql.test.tsx]: i have reverted the code

File: src/JsonForm/Fields/FormTextInput.tsx (hunk 7 above)
Comment (adnan1naeem) [reply src/JsonForm/Fields/FormTextInput.tsx]: I have reverted the code

File: src/JsonForm/Fields/FormTextInput.tsx (hunk 6 above)
Comment (adnan1naeem) [reply src/JsonForm/Fields/FormTextInput.tsx]: i have reverted the code

File: src/JsonForm/BaseFormField.tsx (hunk 8 above)
Comment (adnan1naeem) [reply src/JsonForm/BaseFormField.tsx]: i have reverted the code

File: src/JsonForm/Fields/FormTextInput.tsx (hunk 4 above)
Comment (adnan1naeem) [reply src/JsonForm/Fields/FormTextInput.tsx]: I have reverted the code

File: src/JsonForm/Fields/FormTextInput.tsx [hunk 9]
```diff
@@ -44,7 +44,7 @@ export const FormTextInput = (props: FormTextInputProps) => (
 				...field,
 				onChange: undefined,
 				...props,
-				error: (form.errors[name] && form.touched[name]) || props.error,
+				error: form.errors[name] ? true : false || props.error,
```
Comment (artalat) [inline src/JsonForm/Fields/FormTextInput.tsx]:
```suggestion
				error: !!form.errors[name] || props.error,
```

Can you test this instead

**Review by artalat — CHANGES_REQUESTED** [review]

File: src/JsonForm/Fields/FormTextInput.tsx (hunk 9 above)
Comment (adnan1naeem) [reply src/JsonForm/Fields/FormTextInput.tsx]: it worked fixed it

**Review by artalat — APPROVED** [review]

**Review by artalat — APPROVED** [review]

**Pull request merged**